src/analysis/calculus/mean_value.lean

```diff
@@ -895,13 +895,36 @@ begin
   exact hf'_mono a b (hxyD' ⟨hxa, hay⟩) (hyzD' ⟨hyb, hbz⟩) (le_of_lt $ lt_trans hay hyb)
 end
 
+/-- If a function `f` is continuous on a convex set `D ⊆ ℝ`, is differentiable on its interior,
+and `f'` is antimonotone on the interior, then `f` is concave on `D`. -/
+theorem concave_on_of_deriv_antimono {D : set ℝ} (hD : convex D) {f : ℝ → ℝ}
+  (hf : continuous_on f D) (hf' : differentiable_on ℝ f (interior D))
+  (hf'_mono : ∀ x y ∈ interior D, x ≤ y → deriv f y ≤ deriv f x) :
+  concave_on D f :=
+begin
+  have : ∀ x y ∈ interior D, x ≤ y → deriv (-f) x ≤ deriv (-f) y,
+  { refine λ x y hx hy hxy, _,
+    change deriv (λ z, -f z) x ≤ deriv (λ z, -f z) y,
+    simp_rw [deriv.neg],
+    exact neg_le_neg (hf'_mono x y hx hy hxy) },
+  exact (neg_convex_on_iff D f).mp (convex_on_of_deriv_mono hD
+    (continuous_on.neg hf) (differentiable_on.neg hf') this),
+end
+
 /-- If a function `f` is continuous on a convex set `D ⊆ ℝ`, is differentiable on its interior,
 and `f'` is monotone on the interior, then `f` is convex on `ℝ`. -/
 theorem convex_on_univ_of_deriv_mono {f : ℝ → ℝ} (hf : differentiable ℝ f)
   (hf'_mono : monotone (deriv f)) : convex_on univ f :=
 convex_on_of_deriv_mono convex_univ hf.continuous.continuous_on hf.differentiable_on
   (λ x y _ _ h, hf'_mono h)
 
+/-- If a function `f` is continuous on a convex set `D ⊆ ℝ`, is differentiable on its interior,
+and `f'` is antimonotone on the interior, then `f` is concave on `ℝ`. -/
```

Comment:
This docstring (and the previous) look like a mistaken copy-paste, there's no `D` mentioned in the lemma?

Reply:
Yes that's definitely a copy/paste mistake.